src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -90,15 +89,30 @@ class ActionPlanService(
 
   fun approveActionPlan(id: UUID, user: AuthUser): ActionPlan {
     val actionPlan = getActionPlan(id)
+
+    actionPlan.referral.approvedActionPlan?. let {
+      reassignSessions(it.id, actionPlan)
+    } ?: run {
+      actionPlanSessionsService.createUnscheduledSessionsForActionPlan(actionPlan)
+    }
+
     actionPlan.approvedAt = OffsetDateTime.now()
     actionPlan.approvedBy = authUserRepository.save(user)
 
-    actionPlanSessionsService.createUnscheduledSessionsForActionPlan(actionPlan)
     val approvedActionPlan = actionPlanRepository.save(actionPlan)
     actionPlanEventPublisher.actionPlanApprovedEvent(approvedActionPlan)
     return approvedActionPlan
   }
 
+  private fun reassignSessions(fromActionPlanId: UUID, toActionPlan: ActionPlan) {
+    val sessions = actionPlanSessionRepository.findAllByActionPlanId(fromActionPlanId)
+    sessions.forEach {
+      it.actionPlan = toActionPlan
+      actionPlanSessionRepository.save(it)
+    }
+    actionPlanSessionsService.createUnscheduledSessionsForActionPlan(toActionPlan, sessions.count())
```

Comment:
Could it not be the case that there could already be some unscheduled sessions already after copying them over?